File: egov/egov-egf/src/main/java/com/exilant/eGov/src/domain/Bank.java
/*******************************************************************************
 * eGov suite of products aim to improve the internal efficiency,transparency, 
 *    accountability and the service delivery of the government  organizations.
 * 
 *     Copyright (C) <2015>  eGovernments Foundation
 * 
 *     The updated version of eGov suite of products as by eGovernments Foundation 
 *     is available at http://www.egovernments.org
 * 
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, either version 3 of the License, or
 *     any later version.
 * 
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 * 
 *     You should have received a copy of the GNU General Public License
 *     along with this program. If not, see http://www.gnu.org/licenses/ or 
 *     http://www.gnu.org/licenses/gpl.html .
 * 
 *     In addition to the terms of the GPL license to be adhered to in using this
 *     program, the following additional terms are to be complied with:
 * 
 * 	1) All versions of this program, verbatim or modified must carry this 
 * 	   Legal Notice.
 * 
 * 	2) Any misrepresentation of the origin of the material is prohibited. It 
 * 	   is required that all modified versions of this material be marked in 
 * 	   reasonable ways as different from the original version.
 * 
 * 	3) This license does not grant any rights to any user of the program 
 * 	   with regards to rights under trademark law for use of the trade names 
 * 	   or trademarks of eGovernments Foundation.
 * 
 *   In case of any queries, you can reach eGovernments Foundation at contact@egovernments.org.
 ******************************************************************************/
/*
 * Created on Jul 13, 2006
 *
 * TODO To change the template for this generated file go to
 * Window - Preferences - Java - Code Style - Code Templates
 */
package com.exilant.eGov.src.domain;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.SQLException;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Locale;

import org.apache.log4j.Logger;
import org.egov.infstr.utils.EgovMasterDataCaching;

import com.exilant.eGov.src.common.EGovernCommon;
import com.exilant.exility.common.TaskFailedException;
import com.exilant.exility.updateservice.PrimaryKeyGenerator;

/**
 * @author Lakshmi
 * 
 *         TODO To change the template for this generated type comment go to
 *         Window - Preferences - Java - Code Style - Code Templates
 */
public class Bank {
	private String id = null;
	private String code = null;
	private String name = null;
	private String narration = null;
	private String isActive = "1";
	private String lastModified = null;
	private String created = null;
	private String modifiedBy = null;
	private String type = null;
	private static final Logger LOGGER = Logger.getLogger(Bank.class);
	private TaskFailedException taskExc;
	private SimpleDateFormat sdf = new SimpleDateFormat("dd/MM/yyyy", Locale
			.getDefault());
	private SimpleDateFormat formatter = new SimpleDateFormat("dd-MMM-yyyy",
			Locale.getDefault());
	private String insertQuery;
	private boolean isId = false, isField = false;

	public String getCode() {
		return code;
	}

	public void setCode(String code) {
		this.code = code;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getNarration() {
		return narration;
	}

	public void setNarration(String narration) {
		this.narration = narration;
	}

	public String getIsActive() {
		return isActive;
	}

	public void setIsActive(String isActive) {
		this.isActive = isActive;
	}

	public String getLastModified() {
		return lastModified;
	}

	public void setLastModified(String lastModified) {
		this.lastModified = lastModified;
	}

	public String getCreated() {
		return created;
	}

	public void setCreated(String created) {
		this.created = created;
	}

	public String getModifiedBy() {
		return modifiedBy;
	}

	public void setModifiedBy(String modifiedBy) {
		this.modifiedBy = modifiedBy;
	}

	public String getType() {
		return type;
	}

	public void setType(String type) {
		this.type = type;
	}

	public SimpleDateFormat getFormatter() {
		return formatter;
	}

	public void setFormatter(SimpleDateFormat formatter) {
		this.formatter = formatter;
	}

	public void setId(String id) {
		this.id = id;
	}

	public String getId() {
		return id;
	}

	public String insert(Connection con) throws TaskFailedException,
			SQLException {
		setId(String.valueOf(PrimaryKeyGenerator.getNextKey("Bank")));

		EGovernCommon common = new EGovernCommon();
		try {
			PreparedStatement pstmt = null;

			created = common.getCurrentDate();
			created = formatter.format(sdf.parse(created));
			lastModified = created;
			EgovMasterDataCaching.getInstance().removeFromCache("egf-bank");

			setCreated(created);
			setLastModified(lastModified);
			setCode(id);
			narration=common.formatString(narration);
			insertQuery = "insert into Bank (id,code,Name,Narration,isActive,LastModified,Created,ModifiedBy,Type)"
					+ "values(?,?,?,?,?,?,?,?,?)";
			pstmt = con.prepareStatement(insertQuery);
			pstmt.setString(1, removeSingleQuotes(id));
			pstmt.setString(2, removeSingleQuotes(code));
			pstmt.setString(3, removeSingleQuotes(name));
			pstmt.setString(4, removeSingleQuotes(narration));
			pstmt.setString(5, removeSingleQuotes(isActive));
			pstmt.setString(6, removeSingleQuotes(lastModified));
			pstmt.setString(7, removeSingleQuotes(created));
			pstmt.setString(8, removeSingleQuotes(modifiedBy));
			pstmt.setString(9, removeSingleQuotes(type));
			if(LOGGER.isDebugEnabled())     LOGGER.debug("B4 insertion: " + insertQuery);
			pstmt.executeUpdate();
			pstmt.close();
			if(LOGGER.isDebugEnabled())     LOGGER.debug(insertQuery);
		} catch (Exception e) {
			LOGGER.error("Exp in insert" + e.getMessage(), e);
			throw taskExc;
		} 
		return id;
	}

	private String removeSingleQuotes(String obj) {
		if (obj != null) {
			obj = obj.replaceAll("'", "");
		}
		return obj;

	}

	public void update(Connection con) throws TaskFailedException, SQLException {
		newUpdate(con);
	}

	public void newUpdate(Connection con) throws TaskFailedException,
			SQLException {
		EGovernCommon commommethods = new EGovernCommon();
		created = commommethods.getCurrentDate();
		PreparedStatement pstmt = null;
		try {
			created = formatter.format(sdf.parse(created));
		} catch (ParseException parseExp) {
			if(LOGGER.isDebugEnabled())     LOGGER.debug(parseExp.getMessage(), parseExp);
		}
		setCreated(created);
		setLastModified(created);
		StringBuilder query = new StringBuilder(500);
		query.append("update Bank set ");
		if (code != null)
			query.append("code=?,");
		if (name != null)
			query.append("name=?,");
		if (narration != null)
			query.append("narration=?,");
		if (isActive != null)
			query.append("isActive=?,");
		if (lastModified != null)
			query.append("lastModified=?,");
		if (created != null)
			query.append("created=?,");
		if (modifiedBy != null)
			query.append("modifiedBy=?,");
		if (type != null)
			query.append("type=?,");
		int lastIndexOfComma = query.lastIndexOf(",");
		query.deleteCharAt(lastIndexOfComma);
		query.append(" where id=?");
		try {
			int i = 1;
			pstmt = con.prepareStatement(query.toString());
			if (code != null)
				pstmt.setString(i++, code);
			if (name != null)
				pstmt.setString(i++, name);
			if (narration != null)
				pstmt.setString(i++, narration);
			if (isActive != null)
				pstmt.setString(i++, isActive);
			if (lastModified != null)
				pstmt.setString(i++, lastModified);
			if (created != null)
				pstmt.setString(i++, created);
			if (modifiedBy != null)
				pstmt.setString(i++, modifiedBy);
			if (type != null)
				pstmt.setString(i++, type);
			pstmt.setString(i++, id);

			pstmt.executeQuery();
		} catch (Exception e) {
			LOGGER.error("Exp in update: " + e.getMessage());
			throw taskExc;
		} finally {
			try {
				pstmt.close();
			} catch (Exception e) {
				LOGGER.error("Inside finally block of update");
			}
		}
	}

}

File: egov/egov-egfweb/src/main/webapp/resources/javascript/bank.js
/*
 *    eGov  SmartCity eGovernance suite aims to improve the internal efficiency,transparency,
 *    accountability and the service delivery of the government  organizations.
 *
 *     Copyright (C) 2017  eGovernments Foundation
 *
 *     The updated version of eGov suite of products as by eGovernments Foundation
 *     is available at http://www.egovernments.org
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, either version 3 of the License, or
 *     any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program. If not, see http://www.gnu.org/licenses/ or
 *     http://www.gnu.org/licenses/gpl.html .
 *
 *     In addition to the terms of the GPL license to be adhered to in using this
 *     program, the following additional terms are to be complied with:
 *
 *         1) All versions of this program, verbatim or modified must carry this
 *            Legal Notice.
 *            Further, all user interfaces, including but not limited to citizen facing interfaces, 
 *            Urban Local Bodies interfaces, dashboards, mobile applications, of the program and any 
 *            derived works should carry eGovernments Foundation logo on the top right corner.
 *
 *            For the logo, please refer http://egovernments.org/html/logo/egov_logo.png.
 *            For any further queries on attribution, including queries on brand guidelines, 
 *            please contact contact@egovernments.org
 *
 *         2) Any misrepresentation of the origin of the material is prohibited. It
 *            is required that all modified versions of this material be marked in
 *            reasonable ways as different from the original version.
 *
 *         3) This license does not grant any rights to any user of the program
 *            with regards to rights under trademark law for use of the trade names
 *            or trademarks of eGovernments Foundation.
 *
 *   In case of any queries, you can reach eGovernments Foundation at contact@egovernments.org.
 *
 */
var existingCode = '';
var existingName = '';

function checkUniqueBankCode(obj) {
	document.getElementById('codeuniquecode').style.display = 'none';
	var code = obj.value;
	if (code != "" && code != existingCode) {
		var url = "bank.action?model.code=" + code + "&rnd="+ Math.random() + "&mode=UNQ_CODE";
		var callback = {
				success : function(oResponse) {
					if (oResponse.responseText == "false") {
						document.getElementById('codeuniquecode').style.display = 'inline';
						obj.value = "";
					}
				},
				failure : function(oResponse) {
					bootbox.alert("Server error occurred");
				}
		};
		YAHOO.util.Connect.asyncRequest("GET", url, callback);
	}
}

function checkUniqueBankName(obj) {
	document.getElementById('nameuniquename').style.display = 'none';
	var name = obj.value;
	if (name !== "" && existingName != name) {
		var url = "bank.action?model.name=" + name + "&rnd="+ Math.random() + "&mode=UNQ_NAME";
		var callback = {
				success : function(oResponse) {
					if (oResponse.responseText == "false") {
						document.getElementById('nameuniquename').style.display = 'inline';
						obj.value = "";
					}
				},
				failure : function(oResponse) {
					bootbox.alert("Server error occurred");
				}
		};
		YAHOO.util.Connect.asyncRequest("GET", url, callback);
	}
}

function check_MICR(e) {
	var branchId = jQuery("#id").val();
	var value = jQuery(e.target).val();
	if (value != '') {
		jQuery.ajax({
			url : 'bankBranch.action?mode=CHECK_UNQ_MICR',
			data : {
				branchMICR : value,
				id : branchId
			},
			type : 'POST',
			async : false,
			datatype : 'text',
			success : function(data) {
				if (data == 'false') {
					bootbox.alert('MICR code already exist');
					jQuery(e.target).val("");
				}
			}
		});
	}
}

function check_BankAccounts(branchId) {
	var response;
	jQuery.ajax({
		url : 'bankBranch.action?mode=CHECK_BANK_ACC',
		data : {
			id : branchId
		},
		type : 'POST',
		async : false,
		datatype : 'text',
		success : function(data) {
			if (data == 'true') {
				response  = true;
			}else{
				response  = false;
			}

		}
	});
	return response;
}
function initializeGrid() {
	jQuery("#listsg11")
	.jqGrid(
			{
				caption : "Branch Details",
				url : 'bankBranch.action?mode=LIST_BRANCH&bankId='+ jQuery("#bank_id").val(),
				editurl : 'bankBranch.action?mode=CRUD&bankId='+ jQuery("#bank_id").val(),
				datatype : "json",
				height : 300,
				width : 800,
				hiddengrid : true,
				colNames : [ 'Srl No', 'Branch Name', 'Branch Code','MICR', 'Address', 'Contact Person', 'Phone Number', 'Narration', 'Active' ],
				colModel : [ {name : 'id',index : 'id',key : true,hidden : true,width : 55,	editable : true,editoptions : {readonly : true, size : 10}}, 
				             {name : 'branchname', index : 'branchname',align:'center', width : 90, editable : true, editoptions : {size : 25},editrules : {required : true},formoptions: { elmprefix: "<span class='mandatory1'>*</span>"}}, 
				             {name : 'branchcode', index : 'branchcode',align:'center', width : 90, editable : true, editoptions : {size : 25}, editrules : {required : true},formoptions: { elmprefix: "<span class='mandatory1'>*</span>"}}, 
				             {name : 'branchMICR', index : 'branchMICR',align:'center', width : 90, editable : true, searchoptions: { sopt: ['eq','ne','lt','le','gt','ge', 'in', 'ni'] },editoptions : {size : 25, dataEvents : [ {type : 'blur', fn : check_MICR} ]}}, 
				             {name : 'branchaddress1', index : 'branchaddress1',align:'center', width : 100, sortable : false, editable : true, edittype : "textarea", editoptions : {rows : "2", cols : "20"}, editrules : {required : true},formoptions: { elmprefix: "<span class='mandatory1'>*</span>"}},
				             {name : 'contactperson', index : 'contactperson',align:'center', width : 80, editable : true, editoptions : {size : 25}}, 
				             {name : 'branchphone', index : 'branchphone',align:'center', width : 80, editable : true, editoptions : {size : 25}}, 
				             {name : 'narration', index : 'narration',align:'center', width : 80, sortable : false, editable : true, edittype : "textarea", editoptions : {rows : "2", cols : "20"}}, 
				             {name : 'isActive', index : 'isActive', align:'center',width : 80, editable : true, edittype : "checkbox",searchoptions: { sopt: ['eq','ne']}, editoptions : { value : "Y:N"}} 
				             ],
				             rowNum : 20,
				             rowList : [ 20, 30, 40, 50 ],
				             pager : '#pagersg11',
				             sortname : 'id',
				             viewrecords : true,
				             sortorder : "desc",
				             multiselect : false,
				             subGrid : true,
				             subGridRowExpanded : function(subgrid_id, row_id) {
				            	 var subgrid_table_id, pager_id;
				            	 subgrid_table_id = subgrid_id + "_t";
				            	 pager_id = "p_" + subgrid_table_id;
				            	 jQuery("#" + subgrid_id).html("<table id='"+ subgrid_table_id+ "' class='scroll'></table><div id='"+ pager_id+ "' class='scroll'></div>");
				            	 jQuery("#" + subgrid_table_id)
				            	 .jqGrid(
				            			 {
				            				 caption : "Account Details",
				            				 url : 'bankAccount.action?mode=LIST_BRANCH_ACC&q=2&bankBranchId='+ row_id,
				            				 editurl : 'bankAccount.action?mode=CRUD&bankBranchId='+ row_id,
				            				 colNames : [ 'ID', 'Account No', 'Fund', 'Account Type', 'Description', 'Pay To', 'Usage Type', 'Active' , 'GlCode'],
				            				 colModel : [{name : 'id', index : 'id', key : true, hidden : true, width : 55, editable : true, editoptions : {readonly : true, size : 10}},
				            				             {name : 'accountnumber', index : 'accountnumber',align:'center', width : 80, key : true, editable : true,searchoptions: { sopt: ['eq','ne','lt','le','gt','ge', 'in', 'ni'] }, editoptions : {size : 25}, editrules : { required : true},formoptions: { elmprefix: "<span class='mandatory1'>*</span>"}},
				            				             {name : 'fundname', index : 'fundname',align:'center', width : 130, editable : true, edittype : "select", editoptions : {value : fundJson}, editrules : { required : true},formoptions: { elmprefix: "<span class='mandatory1'>*</span>"}},
				            				             {name : 'accounttype', index : 'accounttype', width : 70, editable :true, edittype : "select", editoptions : {value : accTypeJson},formoptions: { elmprefix: "<span class='mandatory1'>*</span>"}},
				            				             {name : 'narration', index : 'narration',align:'center', width : 70, editable : true, edittype : "textarea", editoptions : { rows : "2", cols : "20" } },
				            				             {name : 'payto', index : 'payto',align:'center', width : 70, editable : true, editoptions : {size : 25}},
				            				             {name : 'typename', index : 'typename',align:'center', width : 70, editable : true, edittype : "select", editoptions : {value : bankAccTypeJson}, editrules : { required : true},formoptions: { elmprefix: "<span class='mandatory1'>*</span>"}},
				            				             {name : 'active', index : 'active',align:'center', width : 70, editable : true, edittype : "checkbox",searchoptions: { sopt: ['eq','ne']}, editoptions : { value : "Y:N"}},
				            				             {name : 'glcode', index : 'glcode', align:'center',key : true, width : 60, editable : true,searchoptions: { sopt: ['eq','ne','lt','le','gt','ge', 'in', 'ni'] }, editoptions : { size : 20},formoptions: { elmprefix: "<span class='mandatory1'>*</span>"}}
				            				             ],
				            				             datatype : "json",
				            				             rowNum : 20,
				            				             width : 700,
				            				             pager : pager_id,
				            				             multiselect : false,
				            				             viewrecords : true,
				            				             sortname : 'id',
				            				             sortorder : "asc",
				            				             height : '100%'
				            			 });
				            	 jQuery("#" + subgrid_table_id).jqGrid('navGrid',"#" + pager_id, 
				            			 {edit : true, add : true, del : false},
				            			 {
				            				 closeAfterEdit:true,
				            				 resize : true,
				            				 editCaption: "Edit Bank Account",
				            				 bSubmit: "Save Bank Account",
				            				 beforeShowForm:function(response,data){
				            					 jQuery("#accounttype").prop('disabled',true);
				            					 jQuery("#glcode").prop('disabled',true);
				            					 jQuery("#glcode").prop('hidden',false);
				            					 jQuery("#tr_glcode").prop('hidden',false);
				            					 jQuery("#accounttype").prop('hidden',false);
				            					 jQuery("#tr_accounttype").prop('hidden',false);
				            				 },
				            				 afterSubmit: function(response,data){
				            					 return afterSubmit(response.responseText,data,"Bank Account Updated ");
				            				 }
				            			 },
				            			 {
				            				 closeAfterAdd:true,
				            				 resize : true,
				            				 addCaption: "Add Bank Account",
				            				 bSubmit: "Save Bank Account",
				            				 beforeShowForm:function(response,data){
				            					 jQuery("#accounttype").prop('disabled',false);
				            					 jQuery("#glcode").prop('disabled',false);
				            					 if(autoBankGLCode == 'true'){
				            						 jQuery("#glcode").prop('hidden',true);
				            						 jQuery("#tr_glcode").prop('hidden',true);
				            						 jQuery("#" + subgrid_table_id).jqGrid('setColProp', 'accounttype', {editrules: {required: true}});
				            					 }
				            					 else{
				            						 jQuery("#accounttype").prop('hidden',true);
				            						 jQuery("#tr_accounttype").prop('hidden',true);
				            						 jQuery("#" + subgrid_table_id).jqGrid('setColProp', 'glcode', {editrules: {required: true}});
				            					 }
				            				 },
				            				 errorTextFormat: function (response) {
				            					 return '<span class="ui-icon ui-icon-alert" ' +
				            					 'style="float:left; margin-right:.3em;"></span>' + response.responseText;
				            				 },
				            				 afterSubmit: function(response,data){
				            					 return afterSubmit(response.responseText,data,"Bank Account Added ");
				            				 }
				            			 },{
				            				 caption: "Remove Bank Account",
				            				 msg: "Remove the selected Bank Account ?"
				            			 });
				             },
				             subGridRowColapsed : function(subgrid_id, row_id) {
				            	 // this function is called before removing the data
				            	 // var subgrid_table_id;
				            	 // subgrid_table_id = subgrid_id+"_t";
				            	 // jQuery("#"+subgrid_table_id).remove();
				             }
			});
	jQuery("#listsg11").jqGrid('navGrid', '#pagersg11', 
			{add : true,edit : true, del : false},
			{
				closeAfterEdit:true,
				checkOnUpdate:true,
				checkOnSubmit:true,					
				bSubmit: "Save Bank Branch",
				editCaption: "Edit Bank Branch",
				resize : true,
				afterSubmit: function(response,data){
					return afterSubmit(response.responseText,data,"Bank Branch Updated ");
				}				
			},
			{
				closeAfterAdd:true,
				checkOnUpdate:true,
				bSubmit: "Save Bank Branch",
				addCaption: "Add Bank Branch",
				resize : true,
				afterSubmit: function(response,data){
					return afterSubmit(response.responseText,data,"Bank Branch Added ");
				}
			},{
				caption: "Remove Bank Branch",
				msg: "Remove the selected Bank Branch ?",
				beforeSubmit : function(data, formid) { 
					if(check_BankAccounts(data)){
						return[false,"Should not delete branch!!!"];
					}
					else{
						return[true,"Deleted Bank branch successfully"];
					}


				} 
			},{multipleSearch:true});
}

function viewGrid()
{
	jQuery("#listsg11")
	.jqGrid(
			{
				caption : "Branch Details",
				url : 'bankBranch.action?mode=LIST_BRANCH&bankId='+ jQuery("#bank_id").val(),
				datatype : "json",
				height : 300,
				width : 800,
				hiddengrid : true,
				colNames : [ 'Srl No', 'Branch Name', 'Branch Code','MICR', 'Address', 'Contact Person', 'Phone Number', 'Narration', 'Active' ],
				colModel : [ {name : 'id',index : 'id',key : true,hidden : true,width : 55,	editable : true,editoptions : {readonly : true, size : 10}}, 
				             {name : 'branchname', index : 'branchname',align:'center', width : 90, editable : true, editoptions : {size : 25},editrules : {required : true},formoptions: { elmprefix: "<span class='mandatory1'>*</span>"}}, 
				             {name : 'branchcode', index : 'branchcode',align:'center', width : 90, editable : true, editoptions : {size : 25}, editrules : {required : true},formoptions: { elmprefix: "<span class='mandatory1'>*</span>"}}, 
				             {name : 'branchMICR', index : 'branchMICR',align:'center', width : 90, editable : true, searchoptions: { sopt: ['eq','ne','lt','le','gt','ge', 'in', 'ni'] },editoptions : {size : 25, dataEvents : [ {type : 'blur', fn : check_MICR} ]}}, 
				             {name : 'branchaddress1', index : 'branchaddress1',align:'center', width : 100, sortable : false, editable : true, edittype : "textarea", editoptions : {rows : "2", cols : "20"}, editrules : {required : true},formoptions: { elmprefix: "<span class='mandatory1'>*</span>"}},
				             {name : 'contactperson', index : 'contactperson',align:'center', width : 80, editable : true, editoptions : {size : 25}}, 
				             {name : 'branchphone', index : 'branchphone',align:'center', width : 80, editable : true, editoptions : {size : 25}}, 
				             {name : 'narration', index : 'narration',align:'center', width : 80, sortable : false, editable : true, edittype : "textarea", editoptions : {rows : "2", cols : "20"}}, 
				             {name : 'isActive', index : 'isActive', align:'center',width : 80, editable : true, edittype : "checkbox",searchoptions: { sopt: ['eq','ne']}, editoptions : { value : "Y:N"}} 
				             ],
				             rowNum : 20,
				             rowList : [ 20, 30, 40, 50 ],
				             pager : '#pagersg11',
				             sortname : 'id',
				             viewrecords : true,
				             sortorder : "desc",
				             multiselect : false,
				             subGrid : true,
				             subGridRowExpanded : function(subgrid_id, row_id) {
				            	 var subgrid_table_id, pager_id;
				            	 subgrid_table_id = subgrid_id + "_t";
				            	 pager_id = "p_" + subgrid_table_id;
				            	 jQuery("#" + subgrid_id).html("<table id='"+ subgrid_table_id+ "' class='scroll'></table><div id='"+ pager_id+ "' class='scroll'></div>");
				            	 jQuery("#" + subgrid_table_id)
				            	 .jqGrid(
				            			 {
				            				 caption : "Account Details",
				            				 url : 'bankAccount.action?mode=LIST_BRANCH_ACC&q=2&bankBranchId='+ row_id,
				            				 colNames : [ 'ID', 'Account No', 'Fund', 'Account Type', 'Description', 'Pay To', 'Usage Type', 'Active' , 'GlCode'],
				            				 colModel : [{name : 'id', index : 'id', key : true, hidden : true, width : 55, editable : true, editoptions : {readonly : true, size : 10}},
				            				             {name : 'accountnumber', index : 'accountnumber',align:'center', width : 80, key : true, editable : true,searchoptions: { sopt: ['eq','ne','lt','le','gt','ge', 'in', 'ni'] }, editoptions : {size : 25}, editrules : { required : true},formoptions: { elmprefix: "<span class='mandatory1'>*</span>"}},
				            				             {name : 'fundname', index : 'fundname',align:'center', width : 130, editable : true, edittype : "select", editoptions : {value : fundJson}, editrules : { required : true},formoptions: { elmprefix: "<span class='mandatory1'>*</span>"}},
				            				             {name : 'accounttype', index : 'accounttype', width : 70, editable :true, edittype : "select", editoptions : {value : accTypeJson},formoptions: { elmprefix: "<span class='mandatory1'>*</span>"}},
				            				             {name : 'narration', index : 'narration',align:'center', width : 70, editable : true, edittype : "textarea", editoptions : { rows : "2", cols : "20" } },
				            				             {name : 'payto', index : 'payto',align:'center', width : 70, editable : true, editoptions : {size : 25}},
				            				             {name : 'typename', index : 'typename',align:'center', width : 70, editable : true, edittype : "select", editoptions : {value : bankAccTypeJson}, editrules : { required : true},formoptions: { elmprefix: "<span class='mandatory1'>*</span>"}},
				            				             {name : 'active', index : 'active',align:'center', width : 70, editable : true, edittype : "checkbox",searchoptions: { sopt: ['eq','ne']}, editoptions : { value : "Y:N"}},
				            				             {name : 'glcode', index : 'glcode', align:'center',key : true, width : 60, editable : true,searchoptions: { sopt: ['eq','ne','lt','le','gt','ge', 'in', 'ni'] }, editoptions : { size : 20},formoptions: { elmprefix: "<span class='mandatory1'>*</span>"}}
				            				             ],
				            				             datatype : "json",
				            				             rowNum : 20,
				            				             width : 700,
				            				             pager : pager_id,
				            				             multiselect : false,
				            				             viewrecords : true,
				            				             sortname : 'id',
				            				             sortorder : "asc",
				            				             height : '100%'
				            			 });}
				            	 });
	
}
function showMessage (msg) {
	jQuery.jgrid.info_dialog("Info","<div class=\"ui-state-highlight\" style=\"padding:5px;\">"+msg+"!</div>");
	jQuery("#info_dialog").delay(3000).fadeOut();
	jQuery(".ui-widget-overlay").hide();
}

function afterSubmit(reply,data, action) {
	var isSuccess = false;
	var message='';
	if (reply == "success") {
		showMessage(action+"successfully.");
		isSuccess = true;
	} else {
		message = action+"failed.";
	}
	return [isSuccess,message,data.id];
}